src/realm/sync/noinst/client_impl_base.hpp

```diff
@@ -1090,41 +1084,28 @@ class ClientImpl::Session {
     SyncProgress m_progress;
 
     // In general, the local version produced by the last changeset in the local
-    // history. The uploading process will never advance beyond this point. The
-    // changeset that produced this version may, or may not contain changes of
-    // local origin.
+    // history. The changeset that produced this version may, or may not
+    // contain changes of local origin.
     //
     // It is set to the current version of the local Realm at session activation
     // time (although always zero for the initial empty Realm
-    // state). Thereafter, it is generally updated when the application calls
-    // recognize_sync_version() and when changesets are received from the server
-    // and integrated locally.
+    // state). Thereafter, it is updated when the application calls
+    // recognize_sync_version(), when changesets are received from the server
+    // and integrated locally, and when the uploading process discovers newer
+    // versions.
     //
     // INVARIANT: m_progress.upload.client_version <= m_last_version_available
     version_type m_last_version_available = 0;
 
-    // The target version for the upload process. When the upload cursor
-    // (`m_upload_progress`) reaches `m_upload_target_version`, uploading stops.
-    //
-    // In general, `m_upload_target_version` follows `m_last_version_available`
-    // as it is increased, but in some cases, `m_upload_target_version` will be
-    // kept fixed for a while in order to constrain the uploading process.
```

Comment:
I think this was once true, but now the only time m_upload_target_version wasn't the same as m_last_version_available was at times where we couldn't be uploading changesets anyway (such as while in the process of applying a client reset recovery on the sync worker thread).